test/framework/toolchain.py

```diff
@@ -339,6 +353,20 @@ def test_optarch_generic(self):
                         else:
                             self.assertFalse(generic_flags in tc.get_variable(var))
 
+    def test_optarch_aarch64_heuristic(self):
```

Comment:
@geimer Hmm, you should get a fresh instance... Do make sure you also call `tc.set_options()` before `tc.prepare()` though, even if you're not setting any custom toolchain options. Maybe that explains it?